src/javaharness/java/arcs/api/StorageProxy.java

```diff
@@ -28,10 +29,13 @@ public void register(NativeParticle particle, Handle handle) {
     observers.put(handle, particle);
 
     if (!listenerAttached) {
-      port.InitializeProxy(this); // TODO: pass `onSynchronize` callback to the port.
+      port.InitializeProxy(this, new Function<PortableJson, Void>() {
+          @Override public Void apply(PortableJson json) {
```

Comment:
If you expect a method to take a parameter, but return nothing, don't use Function<Param, Result>, but use Consumer<ParamType> in the API (e.g. initializeProxy(Consumer<T>) }

Reply:
done, thanks!